### DISCUSSION TITLE
Inline-completions with js object with as a resource

### DISCUSSION THREAD

**Discussion by moranmono** [microsoft/vscode-discussions#679] [Extension Development QnA] [answered]
Hi
I am trying to build a vs code extension for my company's customer success team. My goal is to provide them with inline code completion for our js object, which they must fill in various places in the client's back office. 

I looked at this example - [inline-completions](https://github.com/microsoft/vscode-extension-samples/tree/main/inline-completions), and I think this is what I am looking for. 
In this example, the inline completions are generated from the `playground.js` file as a text string, but for my use case, I want the completion to be generated by a js object 
ex:
```
const person = {
  firstName: "John",
  lastName: "Doe",
  age: 50,
  eyeColor: "blue"
};
```
So if I type 'person', I will have an auto-complete of all the person's object properties. 

Is there a way to achieve this? 
Am I looking in the right direction or should I use a different vs extension mode?

**Comment by isidorn** [comment, marked as answer]
Yes, you can use InlineCompletions for this. Here are the docs https://code.visualstudio.com/api/references/vscode-api#InlineCompletionItemProvider

If something does not work let us know.

**Reply by moranmono** [reply]
Thanks. 
I saw an option to use a json file as a custom data source for HTML and CSS files - [custom-data](https://code.visualstudio.com/api/extension-guides/custom-data-extension).
Can I do it with a Javascript file as well? 
Or use it in conjunction with the inline-completion API?

**Reply by isidorn** [reply]
I suggest to try it out yourself.
